src/Middleware/DispatchMiddleware.php

```diff
@@ -28,65 +21,10 @@
  * `MarshalMiddlewareTrait::prepareMiddleware()` method. In each case, it then
  * processes the middleware.
  *
+ * @deprecated since 2.2.0. This class is now a part of zend-expressive-router,
+ *     and will be removed for the 3.0.0 release.
  * @internal
  */
-class DispatchMiddleware implements ServerMiddlewareInterface
+class DispatchMiddleware extends BaseDispatchMiddleware
 {
-    use MarshalMiddlewareTrait;
-
-    /**
-     * @var ContainerInterface|null
-     */
-    private $container;
-
-    /**
-     * @var ResponseInterface
-     */
-    private $responsePrototype;
-
-    /**
-     * @var RouterInterface
-     */
-    private $router;
-
-    /**
-     * @param RouterInterface $router
-     * @param ResponseInterface $responsePrototype
-     * @param ContainerInterface|null $container
-     */
-    public function __construct(
-        RouterInterface $router,
-        ResponseInterface $responsePrototype,
-        ContainerInterface $container = null
-    ) {
-        $this->router = $router;
-        $this->responsePrototype = $responsePrototype;
-        $this->container = $container;
-    }
-
-    /**
-     * @param ServerRequestInterface $request
-     * @param DelegateInterface $delegate
-     * @return ResponseInterface
-     */
-    public function process(ServerRequestInterface $request, DelegateInterface $delegate)
-    {
-        $routeResult = $request->getAttribute(RouteResult::class, false);
-        if (! $routeResult) {
-            return $delegate->process($request);
-        }
-
-        $middleware = $routeResult->getMatchedMiddleware();
-
-        if (! $middleware instanceof ServerMiddlewareInterface) {
-            $middleware = $this->prepareMiddleware(
-                $middleware,
-                $this->router,
-                $this->responsePrototype,
-                $this->container
-            );
-        }
```

Comment:
This logic was only used previously if using the `ApplicationConfigInjectionTrait`, as it would not first marshal the middleware passed to it before piping our routing it. Since the `ApplicationConfigInjectionDelegator` _does_ do that at this point, this logic would never be hit, allowing it to be removed in the zend-expressive-router variant.